packages/analytics-adapter/src/awaiter.ts

```diff
@@ -0,0 +1,80 @@
+
+export abstract class Awaiter {
+    public interval: number = 50;
+
+    constructor(public readonly maxWaitTime = 5000) {
+        /*
+        In the extending class, call the following in the constructor
+         this.waitForAnalytics();
+         this.waitForEvolv();
+        */
+    }
+
+    abstract getAnalytics(): any;
+    abstract checkAnalyticsProviders(): void; // Optional check for incorrect configuration
+
+    abstract onAnalyticsFound(analytics: any): void;
+    abstract onEvolvFound(): void;
+
+    // Override for customer analytics processor
+    getHandler(): any {
+        return this.getAnalytics();
+    }
+
+    waitForAnalytics() {
+        let analytics = this.getHandler();
+        if (analytics) {
+            this.onAnalyticsFound(analytics);
+            return;
+        }
+
+        const begin = Date.now();
+        const intervalId = setInterval(() => {
+            if ((Date.now() - begin) > this.maxWaitTime) {
+                clearInterval(intervalId);
+                console.log('Evolv: Analytics integration timed out - Couldn\'t find Analytics');
+                this.checkAnalyticsProviders();
+                return;
+            }
+
+            analytics = this.getHandler();
+            if (!analytics) {
+                return;
+            }
+
+            this.onAnalyticsFound(analytics);
+
+            clearInterval(intervalId);
+        }, this.interval);
```

Comment:
[Since you just moved code here from another file, I understand if you're not comfortable changing implementation]

In asset-manager, we might use of `requestAnimationFrame()` instead of `setInterval()` for better performance and responsiveness. See [here](https://github.com/evolv-ai/asset-manager/blob/1c5044f8eb48136e1d674b3a01ee444f6ff71bef/src/helpers.js#L12-L42).